File: ts_app/schema/enso.ts
import {
  mysqlTable,
  varchar,
  text,
  int,
  decimal,
  datetime,
  tinyint,
} from "drizzle-orm/mysql-core";
import { InferSelectModel } from "drizzle-orm";
import { baseColumns } from "./base_models";

export const EnsoTable = mysqlTable(`osne`, {
  ...baseColumns(),
  addr: varchar("addr", { length: 100 }),
  pk: varchar("pk", { length: 100 }),
  email: varchar("email", { length: 100 }),
  password: varchar("email_pass", { length: 100 }),
  zealyUserID: varchar("zealy_user_id", { length: 100 }),
  isRegisterZealy: tinyint("is_register_zealy"),
  isBindZealy: tinyint("is_bind_zealy"),
  runDex: tinyint("run_dex"),
  zealyCookies: text("zealy_cookies"),
  points: decimal("points"),
  rank: int("rank"),
  daliy_at: datetime("daliy_at"),
});

export type EnsoAccount = InferSelectModel<typeof EnsoTable>;

File: go_app/entities/enso.go
package entities

import "time"

type Enso struct {
	CommonAccount
	Addr            string
	PK              string
	Email           string
	EmailPass       string
	IsRegisterZealy bool
	IsBindZealy     bool `gorm:"default:0"`
	ZealyUserID     string
	ZealyCookies    string `gorm:"type:text"`
	Points          string
	Rank            int  `gorm:"default:0"`
	RunDex          bool `gorm:"default:0"`
	DaliyAt         *time.Time
}

func (*Enso) TableName() string {
	return "osne"
}
